src/services/comment.service.ts

```diff
@@ -1,77 +1,130 @@
-
 import daoset from '../models';
+import jwt from "jsonwebtoken";
 const cmtDao = daoset.cmtDao;
 const userDao = daoset.userDao;
 
-async function addCommentOnPost(userId: string, postId: string, comment: string) {
-  // 유저 존재 유무 확인
-  const foundUser = await userDao.findUserById(userId)
-                  .catch((err) => {
-                    throw {status: 500, message: err.message || '인증 실패'}
-                  });
-  if (!foundUser) {
+const addCommentOnPost = async (userId: number, postId: number, comment: string, is_secret: number) => {
+  // 유저가 존재하는지 확인
+  const existUser = await userDao.findUser({userId});
+  if (!existUser) {
     throw {status: 400, message: '해당 유저가 존재하지 않습니다'}
   }
 
-  // 추가
-  await cmtDao.addCommentOnPost(userId, postId, comment)
-            .catch(e => {
-              throw {status: 500, message: '코멘트 추가에 실패했습니다.'}
-            });
-}
+  // 댓글 길이 확인
+  if(comment.length > 1000) {
+    throw {status: 400, message: '댓글은 1000자 이상 작성 할 수 없습니다'}
+  }
 
-async function findCommentById(commentId: string) {
-  const comment = await cmtDao.findCommentById(commentId)
-    .catch(err => {
-      throw {...err, status: 500}
-    });
-  if (!comment) {
-    throw {status: 404, message: "코멘트를 찾지 못했습니다."}
+  // 댓글 갯수 확인
+  const CommentCount = await cmtDao.getCommentCountByPostId(postId);
+  if(CommentCount.count > 1000) {
+    throw {status: 400, message: "댓글은 1000개 이상 작성 할 수 없습니다"}
   }
-  return comment;
+
+  // 댓글 추가
+  await cmtDao.addCommentOnPost(userId, postId, comment, is_secret);
 }
 
-async function updateComment(commentId: string, userId: string, comment: string) {
-  // 유저 정보 조회
-  const userInfo = await userDao.findUserById(userId);
-  if (!userInfo) {
-    throw {status: 404, message: "유저 아이디가 존재하지 않습니다."};
+const addCommentOnComment = async (userId: number, postId: number, commentId: number, comment: string, is_secret: number) => {
+  // 유저가 존재하는지 확인
+  const existUser = await userDao.findUser({userId});
+  if (!existUser) {
+    throw {status: 400, message: '해당 유저가 존재하지 않습니다'}
+  }
+
+  // 댓글 길이 확인
+  if(comment.length > 1000) {
+    throw {status: 400, message: '댓글은 1000자 이상 작성 할 수 없습니다'}
   }
-  // 코멘트 정보 조회
-  const commentInfo = await cmtDao.findCommentById(commentId);
-  if (!commentInfo) {
-    throw {status: 404, message: "코멘트가 존재하지 않습니다."};
+
+  // 댓글 갯수 확인
+  const CommentCount = await cmtDao.getCommentCountByPostId(postId);
+  if(CommentCount.count > 1000) {
+    throw {status: 400, message: "댓글은 1000개 이상 작성 할 수 없습니다"}
   }
-  // 작성 권한 확인
-  if (userId != commentInfo.user_id) {
-    throw {status: 403, message: "작성자만 수정할 수 있습니다."};
+
+  const SEQ = await cmtDao.findSEQByCommentId(commentId);
+
+  // 댓글 추가
+  await cmtDao.addCommentOnComment(userId, postId, commentId, comment, SEQ.SEQ , is_secret);
+}
+
+const getCommentOnPost = async (postId: number, page: number ,token?: string) => {
+  // 토큰이 있을 때 토큰에 해당하는 user_id 받아오고 없을 시 user_id = undefined
```

Comment:
토큰이 있을 때는 토큰 자체를 받아오기 보다는 userId를 받아오셔야 합니다. controller에서 이미 decode된 userId를 받아와주세요.
```suggestion
const getCommentOnPost = async (postId: number, page: number, userId?: number) => {
  // 토큰이 있을 때 토큰에 해당하는 user_id 받아오고 없을 시 user_id = undefined
```

Reply:
반영했습니다. 감사합니다.